blocks/rich-text/index.js

```diff
@@ -398,7 +369,7 @@ export class RichText extends Component {
 
 	onChange() {
 		this.isEmpty = this.editor.dom.isEmpty( this.editor.getBody() );
-		this.savedContent = this.isEmpty ? [] : this.getContent();
+		this.savedContent = this.getContent();
```

Comment:
Any way to prevent the empty check from running twice here?

Reply:
Yes, maybe it could be inferred from the content.